kubernetes-service-catalog 0.1.23 (new formula)

ageekymonk opened the pull request:
This formula is for kubernetes service catalog client called svcat

- [X] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [X] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [X] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

ilovezfs commented on Formula/kubernetes-service-catalog.rb:
```diff
@@ -0,0 +1,30 @@
+      :tag => "v0.1.23",
+      :revision => "87a5db0e1e0359ce372037a235ce4448944c8611"
+
+  depends_on "coreutils" => :build
```
This shouldn't be necessary. Naked calls to `realpath` indicate the software is not properly ported to macOS. Please file an issue upstream.

carolynvs replied:
Since we check in our generated files, you don't need to do any make calls at all in order to do a build for homebrew. The `go build` command you have further down should be sufficient.

Let me know if that doesn't work and I'll help get anything fixed on our side (service catalog), so that we can distribute the svcat client with brew! 💯  I'm super excited about this.

ageekymonk replied:
Awesome @carolynvs. It worked like a charm.

carolynvs commented:
This formula installs the Kubernetes Service Catalog _client_, `svcat`. It may make sense to change the formula name/description to make it clear that this is only the CLI/Client, e.g. `kubernetes-service-catalog-client` or just `svcat` (after the binary name) and make the description to be `Kubernetes Service Catalog Client: Consume services in Kuberentes using the OSB API"`.

What do you think?

ageekymonk commented:
Awesome @carolynvs. It worked like a charm. I also renamed as you suggested to kubernetes-service-catalog-client, updated the description and added `v` to version as well.

carolynvs commented:
I just re-read the [naming guidelines](https://github.com/Homebrew/brew/blob/master/docs/Formula-Cookbook.md#a-quick-word-on-naming) and they say 

> Name the formula like the project markets the product

So I am guessing that the current name "kubernetes-service-catalog-client" is most appropriate? I'm sure @ilovezfs will let us know if it should just be named after the binary `svcat` instead.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,28 @@
+    dir.install buildpath.children
+
+    cd dir do
+      ldflags = "-s -w -X github.com/kubernetes-incubator/service-catalog/pkg.VERSION=v#{version}"
```
This should be wrapped to 80 characters, ideally.

ageekymonk replied:
Thanks @DomT4 . I tried to bring it below 80 characters but the url is too long for that. The only way would be for me to break the url and would not be good. Let me know if it is okay to just remain as such.

ageekymonk replied:
Thanks @DomT4. Changed it as requested. Nice idea.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,28 @@
+    cd dir do
+      ldflags = "-s -w -X github.com/kubernetes-incubator/service-catalog/pkg.VERSION=v#{version}"
+      system "go", "build", "-ldflags", ldflags, "-o", "bin/svcat", "./cmd/svcat"
+      bin.install "bin/svcat"
```
You'll likely need to add a `prefix.install_metafiles` line here.

This ensures things like `README`, `LICENSE` and `CHANGELOG` where available are installed into the prefix. Normally Homebrew handles this automatically with a little magic, but when you relocate the entire contents of the `buildpath` to another directory for the install process as is common with `go`-using formulae that logic falls on its face & it needs a little manual nudging. It's a known issue.

ageekymonk replied:
Thanks @DomT4 . Added it.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,28 @@
+
+  test do
+    version_output = shell_output("#{bin}/svcat version --client 2>&1")
+    assert_match "Client Version: v0.1.23", version_output
```
Don't hardcode the version here. Use `#{version}`, but ideally something that tests a fuller level of functionality would be nice.

ageekymonk replied:
Thanks @DomT4 . Fixed it.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,33 @@
+  desc "Consume Services in k8s using the OSB API"
+  homepage "https://svc-cat.io/"
+  url "https://github.com/kubernetes-incubator/service-catalog.git",
+      :tag => "v0.1.23",
```
Looks like we're up to `0.1.28` now, if you wouldn't mind catching that.

ageekymonk replied:
Updated to 0.1.28. Thanks for being thorough.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,33 @@
+    ENV["GOPATH"] = buildpath
+    ENV["NO_DOCKER"] = "1"
+    dir = buildpath/"src/github.com/kubernetes-incubator/service-catalog"
+
```
Move `dir = buildpath/"src/github.com/kubernetes-incubator/service-catalog"` down one line & `dir.install buildpath.children` up one line. This is nit-picking but since other changes are being requested figured it's hopefully not too annoying to ask.

ageekymonk replied:
Updated it. Code readability is important. Thanks for being thorough.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,33 @@
+        github.com/kubernetes-incubator/service-catalog/pkg.VERSION=v#{version}
+      ]
+      system "go", "build", "-ldflags", ldflags.join(" "), "-o",
+             "bin/svcat", "./cmd/svcat"
```
If you change this to `bin/"svcat"` you should be able to get rid of the `bin.install "bin/svcat"` below?

ageekymonk replied:
I did not know that. Thanks for letting me know. I have updated it.

DomT4 commented on Formula/kubernetes-service-catalog-client.rb:
```diff
@@ -0,0 +1,33 @@
+  end
+
+  test do
+    version_output = shell_output("#{bin}/svcat version --client 2>&1")
```
Any chance more can be done than a simple version check here?

carolynvs replied:
All of svcat's other commands try to hit a live Kubernetes cluster, so I'm afraid not! 😝

DomT4 commented:
Merged in https://github.com/Homebrew/homebrew-core/commit/edb4344e662ba27dbc5c607648b02d518bf7ab2f. Thank you for your contribution to Homebrew @ageekymonk; we appreciate it! :smiley_cat:

Thanks to @carolynvs for the assist here as well!
